test/integration/targets/mongodb_replicaset/tasks/main.yml

```diff
@@ -0,0 +1,470 @@
+# test code for the mongodb_replicaset module
+# (c) 2019,  Rhys Campbell <rhys.james.campbell@googlemail.com>
+
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+# ============================================================
+
+- name: Ensure tests home exists
+  file:
+    path: "{{ remote_tmp_dir }}/tests"
+    state: directory
+
+- include_tasks: mongod_teardown.yml
+
+- set_fact:
+    current_replicaset: "{{ mongodb_replicaset1 }}"
+
+- include_tasks: mongod_replicaset.yml
+
+# test with yaml list
+- name: Create replicaset with module
+  mongodb_replicaset:
+    login_user: admin
+    login_password: secret
+    login_host: "localhost"
+    login_port: 3001
+    login_database: "admin"
+    replica_set: "{{ mongodb_replicaset1 }}"
+    members:
+     - "localhost:3001"
+     - "localhost:3002"
+     - "localhost:3003"
+
+- name: Ensure is_primary script exists on host
+  copy:
+    src: js/is_primary.js
+    dest: "{{ remote_tmp_dir }}/tests/is_primary.js"
+
+- name: Get replicaset info
+  command: mongo admin --eval "rs.status()" --port 3001
+  register: mongo_output
+
+- name: Assert replicaset name is in mongo_output
+  assert:
+    that:
+      - "mongo_output.changed == true"
+      - "'{{ mongodb_replicaset1 }}' in mongo_output.stdout"
+      - "'localhost:3001' in mongo_output.stdout"
+      - "'localhost:3002' in mongo_output.stdout"
+      - "'localhost:3003' in mongo_output.stdout"
+
+- name: Add mongodb admin user
+  mongodb_user:
+    login_host: localhost
+    login_port: 3001
+    replica_set: "{{ mongodb_replicaset1 }}"
+    database: admin
+    name: "{{ mongodb_admin_user }}"
+    password: "{{ mongodb_admin_password }}"
+    roles: ["root"]
+    state: present
+  register: mongo_admin_user
+  when: test_mongo_auth
+
+- name: Murder all mongod processes
+  shell: pkill mongod && sleep 10;
```

Comment:
Instead of killing the process by name and then sleeping, get the pid, kill it and wait for the pid to exit.

You can use the new `pids` module to get the pid and `wait_for` to wait for `/proc/{{ pid }}/status` to go away (since the test only runs on Linux).

Reply:
All sleep bash commands removed. Replaced with wait_for via port.